operator-common/src/main/java/io/strimzi/operator/common/model/PemAuthIdentity.java

```diff
@@ -0,0 +1,113 @@
+ * This consists of an X509 end entity certificate, corresponding private key, and a chain of X509 CA certificates, all in PEM format.
+ */
+public class PemAuthIdentity {
+
+    private final byte[] pemPrivateKey;
+    private final byte[] pemCertificateChainBytes;
+    private final X509Certificate pemCertificateChain;
+    private String secretName;
+    private String secretNamespace;
+
+    /**
+     * @param secret Kubernetes Secret containing the Cluster Operator public and private key
+     * @param secretKey Key in the Kubernetes Secret that is associated with the requested identity
+     */
+    private PemAuthIdentity(Secret secret, String secretKey) {
+        Optional.ofNullable(secret)
+                .map(Secret::getMetadata)
+                .ifPresent(objectMeta -> {
+                    secretName = objectMeta.getName();
+                    secretNamespace = objectMeta.getNamespace();
+                });
+        pemPrivateKey = Util.decodeFromSecret(secret, String.format("%s.key", secretKey));
+        pemCertificateChainBytes = Util.decodeFromSecret(secret, String.format("%s.crt", secretKey));
+        pemCertificateChain = validateCertificateChain(secretKey);
+    }
+
+    /**
+     * Create a new instance of PemAuthIdentity that represents the identity of the
+     * cluster operator during TLS client authentication. This also validates the provided
+     * Secret contains a valid certificate chain.
+     *
+     * @param secret Kubernetes Secret containing the client authentication identity
+     *
+     * @return PemAuthIdentity to use as the client authentication identity during TLS authentication
+     */
+    public static PemAuthIdentity clusterOperator(Secret secret) {
+        return new PemAuthIdentity(secret, "cluster-operator");
+    }
+
+    /**
+     * Create a new instance of PemAuthIdentity that represents the identity of the
+     * entity (i.e. user or topic) operator during TLS client authentication. This also validates
+     * the provided Secret contains a valid certificate chain.
+     *
+     * @param secret Kubernetes Secret containing the client authentication identity
+     *
+     * @return PemAuthIdentity to use as the client authentication identity during TLS authentication
+     */
+    public static PemAuthIdentity entityOperator(Secret secret) {
+        return new PemAuthIdentity(secret, "entity-operator");
+    }
+
+    /**
+     * @return The certificate chain for this authentication identity as a X509Certificate
+     */
+    public X509Certificate pemCertificateChain() {
+        return pemCertificateChain;
+    }
+
+    /**
+     * @return The certificate chain for this authentication identity as a byte array
+     */
+    public byte[] pemCertificateChainBytes() {
+        return pemCertificateChainBytes;
+    }
+
+    /**
+     * @return The certificate chain for this authentication identity as a String
+     */
+    public String pemCertificateChainString() {
```

Comment:
The String type could be used to represent the certificate chain in multiple ways. It could be the PEM representation. But since we're also involved with `Secrets` we sometimes end up encoding that as base64. I think `certificateChainAsPem()` would be fine for the former, and `certificateChainAsBase64Pem()` if it's the latter.